### PR TITLE
bpo-30806 netrc.__repr__() is broken for writing to file

### PR DESCRIPTION
**Pull request by Bezier89**
https://bugs.python.org/issue30806

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

File: Lib/netrc.py [hunk 1]
```diff
@@ -127,10 +127,10 @@ def __repr__(self):
         rep = ""
         for host in self.hosts.keys():
             attrs = self.hosts[host]
-            rep = rep + "machine "+ host + "\n\tlogin " + repr(attrs[0]) + "\n"
+            rep = rep + "machine "+ host + "\n\tlogin " + str(attrs[0]) + "\n"
```
Comment (methane) [inline Lib/netrc.py]:
```
rep += f"machine {host}\n\tlogin {attrs[0]}\n"
```

File: Lib/netrc.py [hunk 2]
```diff
@@ -127,10 +127,10 @@ def __repr__(self):
...
             if attrs[1]:
-                rep = rep + "account " + repr(attrs[1])
-            rep = rep + "\tpassword " + repr(attrs[2]) + "\n"
+                rep = rep + "\taccount " + str(attrs[1]) + "\n"
+            rep = rep + "\tpassword " + str(attrs[2]) + "\n"
```
Comment (methane) [inline Lib/netrc.py]:
```
    rep += f"\taccount {attrs[1]}\n"
rep += f"\tpassword {attrs[2]}\n"
```

**Review by methane — CHANGES_REQUESTED** [review]

File: Lib/netrc.py (hunk 1 above)
Comment (Bezier89) [reply Lib/netrc.py]: Done

File: Lib/netrc.py (hunk 2 above)
Comment (Bezier89) [reply Lib/netrc.py]: Done

**Comment by Bezier89** [comment]
@methane, I updated this to use f-strings. Is there anything else that needs to be done for this to be merged?

**Comment by methane** [comment]
Sorry for delay.
Would you add NEWS file and test?

**Comment by methane** [comment]
See https://devguide.python.org/committing/#what-s-new-and-news-entries for how to create NEWS entry file.

**Comment by Bezier89** [comment]
@methane This is done. Let me know if it looks good!

**Review by methane — APPROVED** [review]

**Comment by methane** [comment]
Thanks.

**Comment by sloria** [comment]
Will this fix be backported to python 2.7? The issue exists there as well.

**Comment by methane** [comment]
@sloria Will and will not.
This pull request need modify to backport to 2.7, since it using f-string.
If you create pull request for 2.7, please mention me.  I'll see it.

**Comment by sloria** [comment]
I needed this fix, so I released a library that addresses this issue (and also adds write functionality): https://github.com/sloria/tinynetrc .

I can't make any promises, but if I get some time, I'll look into backporting to 2.7.

**Comment by sloria** [comment]
@methane OK, I've made a PR for 2.7: https://github.com/python/cpython/pull/4276

**Pull request merged**